src/components/select/multi-select/multi-select.component.js

```diff
@@ -326,11 +327,14 @@ const MultiSelect = React.forwardRef(
       };
     }, [handleGlobalClick]);
 
+    const onFilterChange = useStableCallback(onFilterChangeProp);
+    const isFirstRender = useRef(true);
```

Comment:
since you've now introduced this `isFirstRender` to two separate components, and I just checked and it's already used in 3 others, it's probably worth extracting to a new custom hook so we can just call `const isFirstRender = useIsFirstRender()` inside any component that needs it. (And this could return the actual bool rather than the ref wrapping it.)

Personally I think that's worth doing as part of this PR, especially as you're already introducing a new custom hook - but I don't mind if we keep this PR as it is and create a new task to refactor this.

Reply:
Acutally I change my mind, I've implemented this hook in 2 different ways and each of these implementations has serious drawbacks when compared to just writing that directly in the component.

1. https://usehooks-ts.com/react-hook/use-is-first-render - this solution works as long as you don't use the result in another `useEffect` hook. Reason for that is the fact that the `useEffect` hook inside the `useIsFirstRender` is invoked first, meaning that the returned boolean when referenced in the outside `useEffect` is always `false`

2. Similar implementation as above, but using `useState` - works fine, but introduces unnecessary rerender